scipy/spatial/tests/test_spherical_voronoi.py

```diff
@@ -182,11 +182,19 @@ def test_rank_deficient(self):
     @pytest.mark.parametrize("n", [8, 15, 21])
     @pytest.mark.parametrize("radius", [0.5, 1, 2])
     @pytest.mark.parametrize("center", [(0, 0, 0), (1, 2, 3)])
-    def test_geodesic_input(self, n, radius, center):
-        U = Rotation.random(random_state=0).as_matrix()
+    def test_circular_input(self, n, radius, center):
+        # generate a random circle
+        rng = np.random.RandomState(0)
+        U = Rotation.random(random_state=rng).as_matrix()
+        h = radius * rng.uniform(0.001, 0.999)
+        circle_radius = np.sqrt(radius**2 - h**2)
```

Comment:
right here maybe we could have one case where `circle_radius==radius` to maintain the geodesic probe as well?

Reply:
Good suggestion.

I have introduced a `fraction` parameter to set the height of the circle. A fraction of zero makes `circle_radius==radius`.